// @facturacr/atv-sdk/dist/src/xmlSchemaHeaderMap.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.xmlExtructures = exports.defaultOptions = exports.declaration = exports.ATV_VERSION = void 0;
exports.ATV_VERSION = '4.4';
const BASE_DOMAIN = 'https://cdn.comprobanteselectronicos.go.cr/xml-schemas';
const buildNs = (type) => `${BASE_DOMAIN}/v${exports.ATV_VERSION}/${type}`;
exports.declaration = '<?xml version="1.0" encoding="utf-8"?>';
exports.defaultOptions = {
    attrNodeName: 'attr',
    format: true,
    indentBy: '  '
};
const COMMON_STRUCTURE = {
    'xmlns:ds': 'http://www.w3.org/2000/09/xmldsig#',
    'xmlns:xsd': 'http://www.w3.org/2001/XMLSchema',
    'xmlns:xsi': 'http://www.w3.org/2001/XMLSchema-instance'
};
function buildAttrs(type) {
    return Object.assign({ xmlns: buildNs(type) }, COMMON_STRUCTURE);
}
exports.xmlExtructures = {
    FacturaElectronica: buildAttrs('facturaElectronica'),
    TiqueteElectronico: buildAttrs('tiqueteElectronico'),
    FacturaElectronicaExportacion: buildAttrs('facturaElectronica'),
    NotaCreditoElectronica: buildAttrs('notaCreditoElectronica'),
    NotaDebitoElectronica: buildAttrs('notaCreditoElectronica'),
    MensajeReceptor: buildAttrs('mensajeReceptor'),
    CCE: buildAttrs('mensajeReceptor'),
    CPCE: buildAttrs('mensajeReceptor'),
    RCE: buildAttrs('mensajeReceptor')
};
