// xadesjs/build/esm/xml/date_time.js
import { __decorate, __metadata } from "tslib";
import { XmlContent, XmlElement } from 'xml-core';
import { dateFormat } from '../dateformat.js';
import { XmlXades } from './xml.js';
import { XadesObject } from './xml_base.js';
let XadesDateTime = class XadesDateTime extends XadesObject {
    OnLoadXml(e) {
        if (e.textContent) {
            this.Value = new Date(e.textContent);
        }
    }
    OnGetXml(e) {
        if (this.Format) {
            e.textContent = dateFormat(this.Value, this.Format);
        }
        else {
            e.textContent = this.Value.toISOString();
        }
    }
};
__decorate([
    XmlContent({
        defaultValue: new Date(),
        required: true,
    }),
    __metadata("design:type", Date)
], XadesDateTime.prototype, "Value", void 0);
XadesDateTime = __decorate([
    XmlElement({
        localName: 'XadesDateTime',
        namespaceURI: XmlXades.NamespaceURI,
        prefix: XmlXades.DefaultPrefix,
    })
], XadesDateTime);
export { XadesDateTime };
