// xadesjs/build/esm/xml/signing_certificate_v2.js
import { __decorate, __metadata } from "tslib";
import { XmlAttribute, XmlChildElement, XmlElement, XmlBase64Converter } from 'xml-core';
import { XmlXades } from './xml.js';
import { XadesCollection, XadesObject } from './xml_base.js';
import { DigestAlgAndValueType } from './signing_certificate.js';
let CertV2 = class CertV2 extends XadesObject {
};
__decorate([
    XmlChildElement({
        localName: XmlXades.ElementNames.CertDigest,
        parser: DigestAlgAndValueType,
        required: true,
    }),
    __metadata("design:type", DigestAlgAndValueType)
], CertV2.prototype, "CertDigest", void 0);
__decorate([
    XmlChildElement({
        localName: XmlXades.ElementNames.IssuerSerialV2,
        namespaceURI: XmlXades.NamespaceURI,
        prefix: XmlXades.DefaultPrefix,
        converter: XmlBase64Converter,
        required: false,
    }),
    __metadata("design:type", Uint8Array)
], CertV2.prototype, "IssuerSerial", void 0);
__decorate([
    XmlAttribute({ localName: XmlXades.AttributeNames.URI }),
    __metadata("design:type", String)
], CertV2.prototype, "Uri", void 0);
CertV2 = __decorate([
    XmlElement({
        localName: XmlXades.ElementNames.Cert,
        parser: CertV2,
    })
], CertV2);
export { CertV2 };
let CertIDListV2 = class CertIDListV2 extends XadesCollection {
};
CertIDListV2 = __decorate([
    XmlElement({
        localName: 'CertIDListV2',
        parser: CertV2,
    })
], CertIDListV2);
export { CertIDListV2 };
let SigningCertificateV2 = class SigningCertificateV2 extends CertIDListV2 {
};
SigningCertificateV2 = __decorate([
    XmlElement({ localName: XmlXades.ElementNames.SigningCertificateV2 })
], SigningCertificateV2);
export { SigningCertificateV2 };
