// xadesjs/build/esm/xml/signer_role.js
import { __decorate, __metadata } from "tslib";
import { XmlChildElement, XmlElement } from 'xml-core';
import { Any } from './any.js';
import { EncapsulatedPKIData } from './encapsulated_pki_data.js';
import { XmlXades } from './xml.js';
import { XadesCollection, XadesObject } from './xml_base.js';
let ClaimedRole = class ClaimedRole extends Any {
};
ClaimedRole = __decorate([
    XmlElement({ localName: XmlXades.ElementNames.ClaimedRole })
], ClaimedRole);
export { ClaimedRole };
let ClaimedRoles = class ClaimedRoles extends XadesCollection {
};
ClaimedRoles = __decorate([
    XmlElement({
        localName: XmlXades.ElementNames.ClaimedRoles,
        parser: ClaimedRole,
    })
], ClaimedRoles);
export { ClaimedRoles };
let CertifiedRole = class CertifiedRole extends EncapsulatedPKIData {
};
CertifiedRole = __decorate([
    XmlElement({ localName: XmlXades.ElementNames.CertifiedRole })
], CertifiedRole);
export { CertifiedRole };
let CertifiedRoles = class CertifiedRoles extends XadesCollection {
};
CertifiedRoles = __decorate([
    XmlElement({
        localName: XmlXades.ElementNames.CertifiedRoles,
        parser: CertifiedRole,
    })
], CertifiedRoles);
export { CertifiedRoles };
let SignerRole = class SignerRole extends XadesObject {
};
__decorate([
    XmlChildElement({ parser: ClaimedRoles }),
    __metadata("design:type", ClaimedRoles)
], SignerRole.prototype, "ClaimedRoles", void 0);
__decorate([
    XmlChildElement({ parser: CertifiedRoles }),
    __metadata("design:type", CertifiedRoles)
], SignerRole.prototype, "CertifiedRoles", void 0);
SignerRole = __decorate([
    XmlElement({ localName: XmlXades.ElementNames.SignerRole })
], SignerRole);
export { SignerRole };
