// xadesjs/build/esm/xml/signed_data_object_properties.js
import { __decorate, __metadata } from "tslib";
import { XmlAttribute, XmlChildElement, XmlElement } from 'xml-core';
import { CommitmentTypeIndication } from './commitment_type_indication.js';
import { DataObjectFormat } from './data_object_format.js';
import { XAdESTimeStamp } from './xades_time_stamp.js';
import { XmlXades } from './xml.js';
import { XadesCollection, XadesObject } from './xml_base.js';
let IndividualDataObjectsTimeStamp = class IndividualDataObjectsTimeStamp extends XAdESTimeStamp {
};
IndividualDataObjectsTimeStamp = __decorate([
    XmlElement({ localName: XmlXades.ElementNames.IndividualDataObjectsTimeStamp })
], IndividualDataObjectsTimeStamp);
export { IndividualDataObjectsTimeStamp };
let IndividualDataObjectsTimeStampCollection = class IndividualDataObjectsTimeStampCollection extends XadesCollection {
};
IndividualDataObjectsTimeStampCollection = __decorate([
    XmlElement({
        localName: 'IndividualDataObjectsTimeStampCollection',
        parser: IndividualDataObjectsTimeStamp,
    })
], IndividualDataObjectsTimeStampCollection);
export { IndividualDataObjectsTimeStampCollection };
let AllDataObjectsTimeStamp = class AllDataObjectsTimeStamp extends XAdESTimeStamp {
};
AllDataObjectsTimeStamp = __decorate([
    XmlElement({ localName: XmlXades.ElementNames.AllDataObjectsTimeStamp })
], AllDataObjectsTimeStamp);
export { AllDataObjectsTimeStamp };
let DataObjectFormatCollection = class DataObjectFormatCollection extends XadesCollection {
};
DataObjectFormatCollection = __decorate([
    XmlElement({
        localName: 'DataObjectFormatCollection',
        parser: DataObjectFormat,
    })
], DataObjectFormatCollection);
export { DataObjectFormatCollection };
let CommitmentTypeIndicationCollection = class CommitmentTypeIndicationCollection extends XadesCollection {
};
CommitmentTypeIndicationCollection = __decorate([
    XmlElement({
        localName: 'CommitmentTypeIndicationCollection',
        parser: CommitmentTypeIndication,
    })
], CommitmentTypeIndicationCollection);
export { CommitmentTypeIndicationCollection };
let AllDataObjectsTimeStampCollection = class AllDataObjectsTimeStampCollection extends XadesCollection {
};
AllDataObjectsTimeStampCollection = __decorate([
    XmlElement({
        localName: 'AllDataObjectsTimeStampCollection',
        parser: AllDataObjectsTimeStamp,
    })
], AllDataObjectsTimeStampCollection);
export { AllDataObjectsTimeStampCollection };
let SignedDataObjectProperties = class SignedDataObjectProperties extends XadesObject {
};
__decorate([
    XmlAttribute({
        localName: XmlXades.AttributeNames.Id,
        defaultValue: '',
    }),
    __metadata("design:type", String)
], SignedDataObjectProperties.prototype, "Id", void 0);
__decorate([
    XmlChildElement({
        parser: DataObjectFormatCollection,
        noRoot: true,
    }),
    __metadata("design:type", DataObjectFormatCollection)
], SignedDataObjectProperties.prototype, "DataObjectFormats", void 0);
__decorate([
    XmlChildElement({
        parser: CommitmentTypeIndicationCollection,
        noRoot: true,
    }),
    __metadata("design:type", CommitmentTypeIndicationCollection)
], SignedDataObjectProperties.prototype, "CommitmentTypeIndications", void 0);
__decorate([
    XmlChildElement({
        parser: AllDataObjectsTimeStampCollection,
        noRoot: true,
    }),
    __metadata("design:type", AllDataObjectsTimeStampCollection)
], SignedDataObjectProperties.prototype, "AllDataObjectsTimeStamps", void 0);
__decorate([
    XmlChildElement({
        parser: IndividualDataObjectsTimeStampCollection,
        noRoot: true,
    }),
    __metadata("design:type", IndividualDataObjectsTimeStampCollection)
], SignedDataObjectProperties.prototype, "IndividualDataObjectsTimeStamps", void 0);
SignedDataObjectProperties = __decorate([
    XmlElement({ localName: XmlXades.ElementNames.SignedDataObjectProperties })
], SignedDataObjectProperties);
export { SignedDataObjectProperties };
