// xadesjs/build/cjs/xml/certificate_values.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertificateValues = exports.EncapsulatedX509CertificateCollection = exports.EncapsulatedX509Certificate = exports.OtherCertificateCollection = exports.OtherCertificate = void 0;
const tslib_1 = require("tslib");
const xml_core_1 = require("xml-core");
const any_js_1 = require("./any.js");
const encapsulated_pki_data_js_1 = require("./encapsulated_pki_data.js");
const xml_js_1 = require("./xml.js");
const xml_base_js_1 = require("./xml_base.js");
let OtherCertificate = class OtherCertificate extends any_js_1.Any {
};
exports.OtherCertificate = OtherCertificate;
exports.OtherCertificate = OtherCertificate = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.OtherCertificate })
], OtherCertificate);
let OtherCertificateCollection = class OtherCertificateCollection extends xml_base_js_1.XadesCollection {
};
exports.OtherCertificateCollection = OtherCertificateCollection;
exports.OtherCertificateCollection = OtherCertificateCollection = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({
        localName: 'OtherCertificateCollection',
        parser: OtherCertificate,
    })
], OtherCertificateCollection);
let EncapsulatedX509Certificate = class EncapsulatedX509Certificate extends encapsulated_pki_data_js_1.EncapsulatedPKIData {
};
exports.EncapsulatedX509Certificate = EncapsulatedX509Certificate;
exports.EncapsulatedX509Certificate = EncapsulatedX509Certificate = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.EncapsulatedX509Certificate })
], EncapsulatedX509Certificate);
let EncapsulatedX509CertificateCollection = class EncapsulatedX509CertificateCollection extends xml_base_js_1.XadesCollection {
};
exports.EncapsulatedX509CertificateCollection = EncapsulatedX509CertificateCollection;
exports.EncapsulatedX509CertificateCollection = EncapsulatedX509CertificateCollection = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({
        localName: 'EncapsulatedX509CertificateCollection',
        parser: EncapsulatedX509Certificate,
    })
], EncapsulatedX509CertificateCollection);
let CertificateValues = class CertificateValues extends xml_base_js_1.XadesObject {
};
exports.CertificateValues = CertificateValues;
tslib_1.__decorate([
    (0, xml_core_1.XmlAttribute)({
        localName: xml_js_1.XmlXades.AttributeNames.Id,
        defaultValue: '',
    }),
    tslib_1.__metadata("design:type", String)
], CertificateValues.prototype, "Id", void 0);
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({
        parser: EncapsulatedX509CertificateCollection,
        noRoot: true,
    }),
    tslib_1.__metadata("design:type", EncapsulatedX509CertificateCollection)
], CertificateValues.prototype, "EncapsulatedX509Certificates", void 0);
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({
        parser: OtherCertificateCollection,
        noRoot: true,
    }),
    tslib_1.__metadata("design:type", OtherCertificateCollection)
], CertificateValues.prototype, "OtherCertificates", void 0);
exports.CertificateValues = CertificateValues = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.CertificateValues })
], CertificateValues);
