// xadesjs/build/cjs/xml/date_time.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.XadesDateTime = void 0;
const tslib_1 = require("tslib");
const xml_core_1 = require("xml-core");
const dateformat_js_1 = require("../dateformat.js");
const xml_js_1 = require("./xml.js");
const xml_base_js_1 = require("./xml_base.js");
let XadesDateTime = class XadesDateTime extends xml_base_js_1.XadesObject {
    OnLoadXml(e) {
        if (e.textContent) {
            this.Value = new Date(e.textContent);
        }
    }
    OnGetXml(e) {
        if (this.Format) {
            e.textContent = (0, dateformat_js_1.dateFormat)(this.Value, this.Format);
        }
        else {
            e.textContent = this.Value.toISOString();
        }
    }
};
exports.XadesDateTime = XadesDateTime;
tslib_1.__decorate([
    (0, xml_core_1.XmlContent)({
        defaultValue: new Date(),
        required: true,
    }),
    tslib_1.__metadata("design:type", Date)
], XadesDateTime.prototype, "Value", void 0);
exports.XadesDateTime = XadesDateTime = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({
        localName: 'XadesDateTime',
        namespaceURI: xml_js_1.XmlXades.NamespaceURI,
        prefix: xml_js_1.XmlXades.DefaultPrefix,
    })
], XadesDateTime);
