// xadesjs/build/cjs/xml/object_identifier.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectIdentifier = exports.DocumentationReferences = exports.DocumentationReference = exports.Identifier = void 0;
const tslib_1 = require("tslib");
const xml_core_1 = require("xml-core");
const xml_js_1 = require("./xml.js");
const xml_base_js_1 = require("./xml_base.js");
let Identifier = class Identifier extends xml_base_js_1.XadesObject {
};
exports.Identifier = Identifier;
tslib_1.__decorate([
    (0, xml_core_1.XmlAttribute)({ localName: xml_js_1.XmlXades.AttributeNames.Qualifier }),
    tslib_1.__metadata("design:type", String)
], Identifier.prototype, "Qualifier", void 0);
tslib_1.__decorate([
    (0, xml_core_1.XmlContent)({
        defaultValue: '',
        required: true,
    }),
    tslib_1.__metadata("design:type", String)
], Identifier.prototype, "Value", void 0);
exports.Identifier = Identifier = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.Identifier })
], Identifier);
let DocumentationReference = class DocumentationReference extends xml_base_js_1.XadesObject {
    OnLoadXml(e) {
        if (e.textContent) {
            this.Uri = e.textContent;
        }
    }
    OnGetXml(e) {
        if (this.Uri) {
            e.textContent = this.Uri;
        }
    }
};
exports.DocumentationReference = DocumentationReference;
tslib_1.__decorate([
    (0, xml_core_1.XmlContent)({
        defaultValue: '',
        required: true,
    }),
    tslib_1.__metadata("design:type", String)
], DocumentationReference.prototype, "Uri", void 0);
exports.DocumentationReference = DocumentationReference = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.DocumentationReference })
], DocumentationReference);
let DocumentationReferences = class DocumentationReferences extends xml_base_js_1.XadesCollection {
};
exports.DocumentationReferences = DocumentationReferences;
exports.DocumentationReferences = DocumentationReferences = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({
        localName: xml_js_1.XmlXades.ElementNames.DocumentationReferences,
        parser: DocumentationReference,
    })
], DocumentationReferences);
let ObjectIdentifier = class ObjectIdentifier extends xml_base_js_1.XadesObject {
};
exports.ObjectIdentifier = ObjectIdentifier;
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({
        parser: Identifier,
        required: true,
    }),
    tslib_1.__metadata("design:type", Identifier)
], ObjectIdentifier.prototype, "Identifier", void 0);
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({
        localName: xml_js_1.XmlXades.ElementNames.Description,
        namespaceURI: xml_js_1.XmlXades.NamespaceURI,
        prefix: xml_js_1.XmlXades.DefaultPrefix,
        defaultValue: '',
    }),
    tslib_1.__metadata("design:type", String)
], ObjectIdentifier.prototype, "Description", void 0);
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({ parser: DocumentationReferences }),
    tslib_1.__metadata("design:type", DocumentationReferences)
], ObjectIdentifier.prototype, "DocumentationReferences", void 0);
exports.ObjectIdentifier = ObjectIdentifier = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.ObjectIdentifier })
], ObjectIdentifier);
