// xadesjs/build/cjs/xml/revocation_values.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RevocationValues = exports.CRLValues = exports.EncapsulatedCRLValue = exports.OCSPValues = exports.EncapsulatedOCSPValue = exports.OtherValues = exports.OtherValue = void 0;
const tslib_1 = require("tslib");
const xml_core_1 = require("xml-core");
const encapsulated_pki_data_js_1 = require("./encapsulated_pki_data.js");
const xml_js_1 = require("./xml.js");
const xml_base_js_1 = require("./xml_base.js");
let OtherValue = class OtherValue extends encapsulated_pki_data_js_1.EncapsulatedPKIData {
};
exports.OtherValue = OtherValue;
exports.OtherValue = OtherValue = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.OtherValue })
], OtherValue);
let OtherValues = class OtherValues extends xml_base_js_1.XadesCollection {
};
exports.OtherValues = OtherValues;
exports.OtherValues = OtherValues = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({
        localName: xml_js_1.XmlXades.ElementNames.OCSPValues,
        parser: OtherValue,
    })
], OtherValues);
let EncapsulatedOCSPValue = class EncapsulatedOCSPValue extends encapsulated_pki_data_js_1.EncapsulatedPKIData {
};
exports.EncapsulatedOCSPValue = EncapsulatedOCSPValue;
exports.EncapsulatedOCSPValue = EncapsulatedOCSPValue = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.EncapsulatedOCSPValue })
], EncapsulatedOCSPValue);
let OCSPValues = class OCSPValues extends xml_base_js_1.XadesCollection {
};
exports.OCSPValues = OCSPValues;
exports.OCSPValues = OCSPValues = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({
        localName: xml_js_1.XmlXades.ElementNames.OCSPValues,
        parser: EncapsulatedOCSPValue,
    })
], OCSPValues);
let EncapsulatedCRLValue = class EncapsulatedCRLValue extends encapsulated_pki_data_js_1.EncapsulatedPKIData {
};
exports.EncapsulatedCRLValue = EncapsulatedCRLValue;
exports.EncapsulatedCRLValue = EncapsulatedCRLValue = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.EncapsulatedCRLValue })
], EncapsulatedCRLValue);
let CRLValues = class CRLValues extends xml_base_js_1.XadesCollection {
};
exports.CRLValues = CRLValues;
exports.CRLValues = CRLValues = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({
        localName: xml_js_1.XmlXades.ElementNames.CRLValues,
        parser: EncapsulatedCRLValue,
    })
], CRLValues);
let RevocationValues = class RevocationValues extends xml_base_js_1.XadesObject {
};
exports.RevocationValues = RevocationValues;
tslib_1.__decorate([
    (0, xml_core_1.XmlAttribute)({
        localName: xml_js_1.XmlXades.AttributeNames.Id,
        defaultValue: '',
    }),
    tslib_1.__metadata("design:type", String)
], RevocationValues.prototype, "Id", void 0);
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({ parser: CRLValues }),
    tslib_1.__metadata("design:type", CRLValues)
], RevocationValues.prototype, "CRLValues", void 0);
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({ parser: OCSPValues }),
    tslib_1.__metadata("design:type", OCSPValues)
], RevocationValues.prototype, "OCSPValues", void 0);
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({ parser: OtherValues }),
    tslib_1.__metadata("design:type", OtherValues)
], RevocationValues.prototype, "OtherValues", void 0);
exports.RevocationValues = RevocationValues = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.RevocationValues })
], RevocationValues);
