// xadesjs/build/cjs/xml/signing_certificate_v2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SigningCertificateV2 = exports.CertIDListV2 = exports.CertV2 = void 0;
const tslib_1 = require("tslib");
const xml_core_1 = require("xml-core");
const xml_js_1 = require("./xml.js");
const xml_base_js_1 = require("./xml_base.js");
const signing_certificate_js_1 = require("./signing_certificate.js");
let CertV2 = class CertV2 extends xml_base_js_1.XadesObject {
};
exports.CertV2 = CertV2;
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({
        localName: xml_js_1.XmlXades.ElementNames.CertDigest,
        parser: signing_certificate_js_1.DigestAlgAndValueType,
        required: true,
    }),
    tslib_1.__metadata("design:type", signing_certificate_js_1.DigestAlgAndValueType)
], CertV2.prototype, "CertDigest", void 0);
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({
        localName: xml_js_1.XmlXades.ElementNames.IssuerSerialV2,
        namespaceURI: xml_js_1.XmlXades.NamespaceURI,
        prefix: xml_js_1.XmlXades.DefaultPrefix,
        converter: xml_core_1.XmlBase64Converter,
        required: false,
    }),
    tslib_1.__metadata("design:type", Uint8Array)
], CertV2.prototype, "IssuerSerial", void 0);
tslib_1.__decorate([
    (0, xml_core_1.XmlAttribute)({ localName: xml_js_1.XmlXades.AttributeNames.URI }),
    tslib_1.__metadata("design:type", String)
], CertV2.prototype, "Uri", void 0);
exports.CertV2 = CertV2 = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({
        localName: xml_js_1.XmlXades.ElementNames.Cert,
        parser: CertV2,
    })
], CertV2);
let CertIDListV2 = class CertIDListV2 extends xml_base_js_1.XadesCollection {
};
exports.CertIDListV2 = CertIDListV2;
exports.CertIDListV2 = CertIDListV2 = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({
        localName: 'CertIDListV2',
        parser: CertV2,
    })
], CertIDListV2);
let SigningCertificateV2 = class SigningCertificateV2 extends CertIDListV2 {
};
exports.SigningCertificateV2 = SigningCertificateV2;
exports.SigningCertificateV2 = SigningCertificateV2 = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.SigningCertificateV2 })
], SigningCertificateV2);
