// xadesjs/build/cjs/xml/signer_role.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SignerRole = exports.CertifiedRoles = exports.CertifiedRole = exports.ClaimedRoles = exports.ClaimedRole = void 0;
const tslib_1 = require("tslib");
const xml_core_1 = require("xml-core");
const any_js_1 = require("./any.js");
const encapsulated_pki_data_js_1 = require("./encapsulated_pki_data.js");
const xml_js_1 = require("./xml.js");
const xml_base_js_1 = require("./xml_base.js");
let ClaimedRole = class ClaimedRole extends any_js_1.Any {
};
exports.ClaimedRole = ClaimedRole;
exports.ClaimedRole = ClaimedRole = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.ClaimedRole })
], ClaimedRole);
let ClaimedRoles = class ClaimedRoles extends xml_base_js_1.XadesCollection {
};
exports.ClaimedRoles = ClaimedRoles;
exports.ClaimedRoles = ClaimedRoles = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({
        localName: xml_js_1.XmlXades.ElementNames.ClaimedRoles,
        parser: ClaimedRole,
    })
], ClaimedRoles);
let CertifiedRole = class CertifiedRole extends encapsulated_pki_data_js_1.EncapsulatedPKIData {
};
exports.CertifiedRole = CertifiedRole;
exports.CertifiedRole = CertifiedRole = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.CertifiedRole })
], CertifiedRole);
let CertifiedRoles = class CertifiedRoles extends xml_base_js_1.XadesCollection {
};
exports.CertifiedRoles = CertifiedRoles;
exports.CertifiedRoles = CertifiedRoles = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({
        localName: xml_js_1.XmlXades.ElementNames.CertifiedRoles,
        parser: CertifiedRole,
    })
], CertifiedRoles);
let SignerRole = class SignerRole extends xml_base_js_1.XadesObject {
};
exports.SignerRole = SignerRole;
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({ parser: ClaimedRoles }),
    tslib_1.__metadata("design:type", ClaimedRoles)
], SignerRole.prototype, "ClaimedRoles", void 0);
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({ parser: CertifiedRoles }),
    tslib_1.__metadata("design:type", CertifiedRoles)
], SignerRole.prototype, "CertifiedRoles", void 0);
exports.SignerRole = SignerRole = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.SignerRole })
], SignerRole);
