// xadesjs/build/cjs/xml/signed_data_object_properties.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SignedDataObjectProperties = exports.AllDataObjectsTimeStampCollection = exports.CommitmentTypeIndicationCollection = exports.DataObjectFormatCollection = exports.AllDataObjectsTimeStamp = exports.IndividualDataObjectsTimeStampCollection = exports.IndividualDataObjectsTimeStamp = void 0;
const tslib_1 = require("tslib");
const xml_core_1 = require("xml-core");
const commitment_type_indication_js_1 = require("./commitment_type_indication.js");
const data_object_format_js_1 = require("./data_object_format.js");
const xades_time_stamp_js_1 = require("./xades_time_stamp.js");
const xml_js_1 = require("./xml.js");
const xml_base_js_1 = require("./xml_base.js");
let IndividualDataObjectsTimeStamp = class IndividualDataObjectsTimeStamp extends xades_time_stamp_js_1.XAdESTimeStamp {
};
exports.IndividualDataObjectsTimeStamp = IndividualDataObjectsTimeStamp;
exports.IndividualDataObjectsTimeStamp = IndividualDataObjectsTimeStamp = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.IndividualDataObjectsTimeStamp })
], IndividualDataObjectsTimeStamp);
let IndividualDataObjectsTimeStampCollection = class IndividualDataObjectsTimeStampCollection extends xml_base_js_1.XadesCollection {
};
exports.IndividualDataObjectsTimeStampCollection = IndividualDataObjectsTimeStampCollection;
exports.IndividualDataObjectsTimeStampCollection = IndividualDataObjectsTimeStampCollection = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({
        localName: 'IndividualDataObjectsTimeStampCollection',
        parser: IndividualDataObjectsTimeStamp,
    })
], IndividualDataObjectsTimeStampCollection);
let AllDataObjectsTimeStamp = class AllDataObjectsTimeStamp extends xades_time_stamp_js_1.XAdESTimeStamp {
};
exports.AllDataObjectsTimeStamp = AllDataObjectsTimeStamp;
exports.AllDataObjectsTimeStamp = AllDataObjectsTimeStamp = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.AllDataObjectsTimeStamp })
], AllDataObjectsTimeStamp);
let DataObjectFormatCollection = class DataObjectFormatCollection extends xml_base_js_1.XadesCollection {
};
exports.DataObjectFormatCollection = DataObjectFormatCollection;
exports.DataObjectFormatCollection = DataObjectFormatCollection = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({
        localName: 'DataObjectFormatCollection',
        parser: data_object_format_js_1.DataObjectFormat,
    })
], DataObjectFormatCollection);
let CommitmentTypeIndicationCollection = class CommitmentTypeIndicationCollection extends xml_base_js_1.XadesCollection {
};
exports.CommitmentTypeIndicationCollection = CommitmentTypeIndicationCollection;
exports.CommitmentTypeIndicationCollection = CommitmentTypeIndicationCollection = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({
        localName: 'CommitmentTypeIndicationCollection',
        parser: commitment_type_indication_js_1.CommitmentTypeIndication,
    })
], CommitmentTypeIndicationCollection);
let AllDataObjectsTimeStampCollection = class AllDataObjectsTimeStampCollection extends xml_base_js_1.XadesCollection {
};
exports.AllDataObjectsTimeStampCollection = AllDataObjectsTimeStampCollection;
exports.AllDataObjectsTimeStampCollection = AllDataObjectsTimeStampCollection = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({
        localName: 'AllDataObjectsTimeStampCollection',
        parser: AllDataObjectsTimeStamp,
    })
], AllDataObjectsTimeStampCollection);
let SignedDataObjectProperties = class SignedDataObjectProperties extends xml_base_js_1.XadesObject {
};
exports.SignedDataObjectProperties = SignedDataObjectProperties;
tslib_1.__decorate([
    (0, xml_core_1.XmlAttribute)({
        localName: xml_js_1.XmlXades.AttributeNames.Id,
        defaultValue: '',
    }),
    tslib_1.__metadata("design:type", String)
], SignedDataObjectProperties.prototype, "Id", void 0);
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({
        parser: DataObjectFormatCollection,
        noRoot: true,
    }),
    tslib_1.__metadata("design:type", DataObjectFormatCollection)
], SignedDataObjectProperties.prototype, "DataObjectFormats", void 0);
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({
        parser: CommitmentTypeIndicationCollection,
        noRoot: true,
    }),
    tslib_1.__metadata("design:type", CommitmentTypeIndicationCollection)
], SignedDataObjectProperties.prototype, "CommitmentTypeIndications", void 0);
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({
        parser: AllDataObjectsTimeStampCollection,
        noRoot: true,
    }),
    tslib_1.__metadata("design:type", AllDataObjectsTimeStampCollection)
], SignedDataObjectProperties.prototype, "AllDataObjectsTimeStamps", void 0);
tslib_1.__decorate([
    (0, xml_core_1.XmlChildElement)({
        parser: IndividualDataObjectsTimeStampCollection,
        noRoot: true,
    }),
    tslib_1.__metadata("design:type", IndividualDataObjectsTimeStampCollection)
], SignedDataObjectProperties.prototype, "IndividualDataObjectsTimeStamps", void 0);
exports.SignedDataObjectProperties = SignedDataObjectProperties = tslib_1.__decorate([
    (0, xml_core_1.XmlElement)({ localName: xml_js_1.XmlXades.ElementNames.SignedDataObjectProperties })
], SignedDataObjectProperties);
