// @facturacr/atv-sdk/dist/src/services/send/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.sendToCustomURL = exports.send = void 0;
const axios_1 = __importDefault(require("axios"));
const MAIN_DOMAIN = 'https://api.comprobanteselectronicos.go.cr/';
const RUT = (process.env.IS_STG) ? 'recepcion-sandbox' : 'recepcion';
const PATH = `${RUT}/v1/recepcion/`;
const SANBOX_URL = 'https://api-sandbox.comprobanteselectronicos.go.cr/recepcion/v1/recepcion';
const URL = (process.env.IS_STG) ? SANBOX_URL : MAIN_DOMAIN + PATH;
function send(token, postOptions) {
    return (0, axios_1.default)({
        url: URL,
        method: 'post',
        data: postOptions,
        headers: {
            Authorization: 'bearer ' + token,
            'Content-Type': 'application/json'
        }
    });
}
exports.send = send;
function sendToCustomURL(token, url) {
    return (0, axios_1.default)({
        url,
        method: 'get',
        headers: {
            Authorization: 'bearer ' + token,
            'Content-Type': 'application/json'
        }
    });
}
exports.sendToCustomURL = sendToCustomURL;
/*
 * https://www.hacienda.go.cr/docs/5d16ade309fe0_Guia_IdP.pdf
 * https://www.hacienda.go.cr/ATV/ComprobanteElectronico/docs/esquemas/2016/v4.2/comprobantes-electronicos-api.html#
 */
